tools/diagnose

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import os
+import platform
+import sys
+import subprocess
+
+from typing import Callable, List
+
+TOOLS_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.dirname(TOOLS_DIR)
+sys.path.insert(0, ROOT_DIR)
+
+def run(check_func):
+    # type: (Callable) -> None
+    '''
+    This decorator simply runs functions.  It makes it more
+    convenient to add new checks without a big main() function.
+    '''
+    rc = check_func()
+    if not rc:
+        sys.exit(1)
+
+def run_command(args):
+    # type: (List[str]) -> None
+    print(' '.join(args))
+    subprocess.check_call(args)
+
+@run
+def check_python_version():
+    # type: () -> bool
+    subprocess.check_call(['/usr/bin/env', 'python', '-V'])
+    return True
+
+@run
+def pwd():
+    # type: () -> bool
+    subprocess.check_call('pwd')
+    return True
+
+@run
+def host_info():
+    # type: () -> bool
+    print(platform.platform())
+    return True
+
+@run
+def check_django():
+    # type: () -> bool
+    import django
+    print('Django version:', django.get_version())
+    return True
+
+@run
+def provision_version():
+    # type: () -> bool
+    fn = os.path.join(ROOT_DIR, 'var/provision_version')
+    with open(fn) as f:
+        version = f.read().strip()
+    print('latest version provisioned:', version)
+    from version import PROVISION_VERSION
+    print('desired version:', PROVISION_VERSION)
+    if version != PROVISION_VERSION:
+        print('You need to provision!')
+        return False
+    return True
+
+@run
+def node_stuff():
+    # type: () -> bool
+    print('node version:')
+    subprocess.check_call(['node', '--version'])
+    return True
+
+@run
+def test_models():
+    # type: () -> bool
+    settings_module = "zproject.test_settings"
+    os.environ['DJANGO_SETTINGS_MODULE'] = settings_module
+    import django
+    django.setup()
+    from zerver.models import UserProfile, Realm
+    print('Num realms: ', Realm.objects.count())
+    print('Num users: ', UserProfile.objects.count())
+    return True
+
+@run
+def check_venv():
+    # type: () -> bool
+    path = os.path.join(ROOT_DIR, 'scripts', 'lib', 'hash_reqs.py')
+    cache_dir = '/srv/zulip-venv-cache/'
+    for fn in ['py2_dev.txt', 'py3_dev.txt']:
+        requirements_file = os.path.join(ROOT_DIR, "requirements", fn)
+        output = subprocess.check_output([path, requirements_file], universal_newlines=True)
+        sha1sum = output.split()[0]
+        print(fn, 'venv sha: ', sha1sum)
+        if not os.path.exists(os.path.join(cache_dir, sha1sum)):
+            print('Your venv may be improperly installed!')
+            return False
+    command = ['ls', '-1', cache_dir]
+    run_command(command)
+    return True
```

Comment:
I think printing them all via `ls` is probably overkill; for me that prints like 100 lines of output :)

I think the main thing that's of interest is whether their environment matches Git (in terms of whether /srv/zulip-venv and /srv/zulip-py3-venv, which are what are actually being used, point at what requirements files suggest).